pallets/collator-selection/src/tests.rs

```diff
@@ -227,18 +240,17 @@ fn session_management_works() {
 		assert_eq!(SessionChangeBlock::get(), 0);
 		assert_eq!(SessionHandlerCollators::get(), vec![1, 2]);
 
-		// add a new collator
-		assert_ok!(CollatorSelection::register_as_candidate(Origin::signed(3)));
```

Comment:
why did this have to change?

Reply:
ahhh so because of the booting It changed your tests significantly because now the adding the extra collator in was being booted before the bottom half of the tests, so I was like hmm if I just add it higher with another candidate this would keep the same spirit of your tests and deal with the removal

Reply:
What about make booting optional, through either of:

- its type would be `Get<Option<_>>` and if it is `None`, we never boot.
- At least in the context of tests, if you want to never boot, we can just set its value to `Bounded::max`

Reply:
hmmm idk if I want to change the logic of the pallet for the tests, I think a less intrusive way is that I create a function that changes the last_block manually to a much higher block so it never boots, then I can put your test back to the original

Reply:
ok after reverting it back turns out after some changes 3 now authors the block first and does not get booted, test reverted back to original, no change needed